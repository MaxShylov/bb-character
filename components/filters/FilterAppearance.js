import { useCallback, useState } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { ButtonAppearance } from './ButtonAppearance';

export const FilterAppearance = ({ onChange }) => {
  const [appearances, setAppearances] = useState([])
  const toggleAppearances = useCallback((number) => {
    let newAppearances = [...appearances]
    if (newAppearances.includes(number)) {
      newAppearances = newAppearances.filter(num => num !== number)
    } else {
      newAppearances.push(number)
    }

    setAppearances(newAppearances)
    onChange(newAppearances)
  }, [appearances, onChange])

  return (
    <View style={styles.wrapper}>
      <Text>Appearance:</Text>

      <ButtonAppearance title="1" onPress={() => toggleAppearances(1)} />
      <ButtonAppearance title="2" onPress={() => toggleAppearances(2)} />
      <ButtonAppearance title="3" onPress={() => toggleAppearances(3)} />
      <ButtonAppearance title="4" onPress={() => toggleAppearances(4)} />
      <ButtonAppearance title="5" onPress={() => toggleAppearances(5)} />
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  buttonWrapper: {
    backgroundColor: '#a3c8fd',
    width: 40,
    height: 40,
    borderRadius: 7,
    marginLeft: 20,
  },
})
