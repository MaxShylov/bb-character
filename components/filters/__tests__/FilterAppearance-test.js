import renderer from 'react-test-renderer';
import { FilterAppearance } from '../FilterAppearance';

test('renders correctly', () => {
  const tree = renderer.create(<FilterAppearance />).toJSON();

  expect(tree).toMatchSnapshot();
});
