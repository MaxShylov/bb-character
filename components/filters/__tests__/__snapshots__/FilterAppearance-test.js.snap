// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text>
    Appearance:
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          4
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 7,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "width": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
  </View>
</View>
`;
